rest/config.go

```diff
@@ -827,13 +828,75 @@ func setupServerContext(config *StartupConfig, persistentConfig bool) (*ServerCo
 		base.Infof(base.KeyAll, "successfully connected to cluster")
 		sc.bootstrapConnection = cluster
 
-		// TODO: CBG-1457 Synchronously find configs in buckets
-		// sc.fetchConfigs()
+		if err := sc.fetchConfigs(); err != nil {
+			return nil, err
+		}
 	}
 
 	return sc, nil
 }
 
+func (sc *ServerContext) fetchConfigs() error {
+	locations, err := sc.bootstrapConnection.GetConfigLocations()
+	if err != nil {
+		return fmt.Errorf("couldn't get buckets from cluster: %w", err)
+	}
+
+	for _, location := range locations {
+		base.Tracef(base.KeyAll, "Checking %q for Sync Gateway config in group %q", location, sc.config.Bootstrap.ConfigGroupID)
+		// populate values from bootstrap, before we overwrite them with any values fetched from the bucket.
```

Comment:
What happens in the situation where the bootstrap is using basic auth, and the db-specific is using x509 (or vice versa)?  Will the bootstrap values be zeroed out correctly?

Reply:
Good catch... They won't be. Updated so that _any_ auth fields defined on a DbConfig overwrite all of the bootstrap auth fields.